client/cloudSubnet_service.go

```diff
@@ -64,33 +82,54 @@ func (sm *ServiceManager) DeleteCloudSubnet(ip string, cloud_cidr_pool_addr stri
 	return sm.DeleteByDn(dn, models.CloudsubnetClassName)
 }
 
-func (sm *ServiceManager) UpdateCloudSubnet(ip string, cloud_cidr_pool_dn string, description string, cloudSubnetattr models.CloudSubnetAttributes, zoneDn string) (*models.CloudSubnet, error) {
+func (sm *ServiceManager) UpdateCloudSubnet(ip string, cloud_cidr_pool_dn string, description string, cloudSubnetattr models.CloudSubnetAttributes, zoneDn, cloudRsSubnetToCtx string) (*models.CloudSubnet, error) {
 	rn := fmt.Sprintf("subnet-[%s]", ip)
-	// parentDn := fmt.Sprintf("uni/tn-%s/ctxprofile-%s/cidr-[%s]", tenant, cloud_context_profile, cloud_cidr_pool_addr)
 	parentDn := cloud_cidr_pool_dn
 	cloudSubnet := models.NewCloudSubnet(rn, parentDn, description, cloudSubnetattr)
 	cloudSubnet.Status = "modified"
 
 	jsonPayload, _, err := sm.PrepareModel(cloudSubnet)
+	jsonPayload.Array(cloudSubnet.ClassName, "children")
 
-	rsZoneAttachJSON := []byte(fmt.Sprintf(`
-	{
-		"cloudRsZoneAttach": {
-			"attributes": {
-				"annotation": "orchestrator:terraform",
-				"dn": "%s/%s/rszoneAttach",
-				"tDn": "%s"
+	if zoneDn != "" {
+		rsZoneAttachJSON := []byte(fmt.Sprintf(`
+		{
+			"cloudRsZoneAttach": {
+				"attributes": {
+					"annotation": "orchestrator:terraform",
+					"dn": "%s/%s/rszoneAttach",
+					"tDn": "%s"
+				}
 			}
 		}
-	}
-	`, parentDn, rn, zoneDn))
-	zoneCon, err := container.ParseJSON(rsZoneAttachJSON)
-	if err != nil {
-		return nil, err
+		`, parentDn, rn, zoneDn))
+		zoneCon, err := container.ParseJSON(rsZoneAttachJSON)
+		if err != nil {
+			return nil, err
+		}
+
+		jsonPayload.ArrayAppend(zoneCon.Data(), cloudSubnet.ClassName, "children")
 	}
 
-	jsonPayload.Array(cloudSubnet.ClassName, "children")
-	jsonPayload.ArrayAppend(zoneCon.Data(), cloudSubnet.ClassName, "children")
+	if cloudRsSubnetToCtx != "" {
+		cloudRsSubnetToCtxJSON := []byte(fmt.Sprintf(`
+		{
+			"cloudRsSubnetToCtx": {
+				"attributes": {
+					"annotation":"orchestrator:terraform",	
```

Comment:
don't hardcode it. Do you have access to the parent orchestrator value?

Reply:
Opened https://github.com/ciscoecosystem/aci-go-client/issues/266 to track this in the future